### DISCUSSION TITLE
Is it possible to configure for default .js file extension (node resolution algorithm)?

### DISCUSSION THREAD

**Discussion by tmpfs** [swc-project/swc#1741] [Q&A] [answered]
I am trying to integrate SWC with an old code base that uses the node resolution algorithm with ES module imports which do not use a `.js` file extension when referencing imports.

Is it possible to configure SWC to support the node module resolution algorithm so that they can be detected. I notice that `spack` has a `NodeResolver` but I cannot find in the docs how to set this as the default for ESM imports.

Thanks for any help.

**Comment by kdy1** [comment]
What happens if you use commonjs module setting?
It would compile your code into cjs modules and as a result it would use the algorithm of node.js.

**Reply by tmpfs** [reply]
Hi @kdy1, how do I configure that please? I am now reading the source to work out the `spack` configuration and am having trouble finding out how to set this up.

**Reply by kdy1** [reply]
Are you using spack?
You can configure it with swcrc, but not for spack.

See https://swc.rs/docs/configuring-swc/

**Reply by tmpfs** [reply]
Edit: yes I am using `spack` to compile, running `npx spack`.

Thanks, I added the `module` declaration with `commonjs` so it looks like this now in `.swcrc`:

```
{
  "module": {
    "type": "commonjs",
    "strict": false,
    "strictMode": true,
    "lazy": false,
    "noInterop": false
  },
  "jsc": {
    "parser": {
      "syntax": "ecmascript",
      "jsx": true,
      "dynamicImport": false,
      "privateMethod": false,
      "functionBind": false,
      "exportDefaultFrom": false,
      "exportNamespaceFrom": false,
      "decorators": false,
      "decoratorsBeforeExport": false,
      "topLevelAwait": false,
      "importMeta": false
    },
    "transform": null,
    "target": "es5",
    "loose": false,
    "externalHelpers": false,
    "keepClassNames": false
  }
}
```

And I still get the same error:

```
(node:11961) UnhandledPromiseRejectionWarning: Error: load_transformed failed

Caused by:
    0: failed to analyze module
    1: failed to resolve ../pages/send/send.utils from /home/muji/git/consensys/metamask-extension/ui/store/actions.js
    2: index not found: /home/muji/git/consensys/metamask-extension/ui/store/../pages/send/send.utils
(Use `node --trace-warnings ...` to show where the warning was created)
(node:11961) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:11961) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Is it possible I am hitting this bug too: https://github.com/swc-project/swc/issues/1076 ?

**Reply by tmpfs** [reply]
I can compile a single module like this `npx swc ui/index.js` but then I can't create a bundle (unless I am missing more options).

**Reply by kumavis** [reply]
@kdy1 are you saying spack doesnt support the commonjs module type?

**Reply by kdy1** [reply]
It does not support **emitting** cjs modules.

**Comment by tmpfs** [comment]
FYI, this problem was related to imports without a valid file extension that contained a period, a fix is in this PR: https://github.com/swc-project/swc/pull/1748.

**Reply by kdy1** [reply]
Thanks! I'll review it after fixing spurious ci failures.

**Comment by tmpfs** [comment, marked as answer]
This problem was fixed with the merge of https://github.com/swc-project/swc/pull/1748.
